Firestore/core/src/remote/serializer.cc

```diff
@@ -1429,9 +1430,29 @@ std::unique_ptr<WatchChange> Serializer::DecodeDocumentRemove(
 
 std::unique_ptr<WatchChange> Serializer::DecodeExistenceFilterWatchChange(
     ReadContext*, const google_firestore_v1_ExistenceFilter& filter) const {
-  ExistenceFilter existence_filter{filter.count};
-  return absl::make_unique<ExistenceFilterWatchChange>(existence_filter,
-                                                       filter.target_id);
+  ExistenceFilter existence_filter = DecodeExistenceFilter(filter);
+  return absl::make_unique<ExistenceFilterWatchChange>(
+      std::move(existence_filter), filter.target_id);
+}
+
+ExistenceFilter Serializer::DecodeExistenceFilter(
+    const google_firestore_v1_ExistenceFilter& filter) const {
+  // Create bloom filter if there is an unchanged_names present in the filter
+  // and inputs are valid, otherwise keep it null.
+  absl::optional<BloomFilter> bloom_filter = absl::nullopt;
+  if (filter.has_unchanged_names && filter.unchanged_names.has_bits) {
+    // TODO(Mila): None of the ported spec tests here actually has the bloom
+    // filter json string, so hard code an empty bloom filter for now. The
+    // actual parsing code will be written in the next PR, where we can validate
+    // the parsing result.
+    StatusOr<BloomFilter> maybe_bloom_filter =
+        BloomFilter::Create(std::vector<uint8_t>{}, 0, 0);
+    if (maybe_bloom_filter.ok()) {
```

Comment:
Since the spec tests in this PR doesn't actually has Bloom filter, and not doing the parsing/serializing, the bloom filter creation is not tested. This part is covered in the next PR.